File: src/main/java/com/extension/factcheck/domain/BaseTimeEntity.java
package com.extension.factcheck.domain;


import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import java.io.Serializable;
import java.time.OffsetDateTime;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class BaseTimeEntity implements Serializable {

  private static final long serialVersionUID = -1039316944866564968L;

  @CreatedDate
  protected OffsetDateTime dateCreated;

  @LastModifiedDate
  protected OffsetDateTime dateUpdated;

  public OffsetDateTime getDateCreated() {
    return dateCreated;
  }

  public OffsetDateTime getDateUpdated() {
    return dateUpdated;
  }
}

File: src/main/kotlin/com/extension/factcheck/domain/BaseTimeEntity.kt
package com.extension.factcheck.domain

import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.io.Serial
import java.io.Serializable
import java.time.LocalDateTime

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class BaseTimeEntity : Serializable {
    @CreatedDate
    lateinit var dateCreated: LocalDateTime

    @LastModifiedDate
    lateinit var dateUpdated: LocalDateTime

    companion object {
        @Serial
        private const val serialVersionUID: Long = -1039316944866564968L
    }
}
